src/Yarp.ReverseProxy.Swagger/ReverseProxyDocumentFilter.cs

```diff
@@ -90,5 +113,72 @@ public void Apply(OpenApiDocument swaggerDoc, DocumentFilterContext context)
             swaggerDoc.SecurityRequirements = securityRequirements;
             swaggerDoc.Components = components;
         }
+
+        private static IReadOnlyDictionary<string, List<string>> GetPublishedPaths(IConfiguration configuration)
+        {
+            var validRoutes = new Dictionary<string, List<string>>();
+            var allConfigs = configuration.AsEnumerable().ToImmutableDictionary();
+            var allPaths = allConfigs.Where(q => q.Key.EndsWith("Match:Path"));
+            foreach (var (key, routeValue) in allPaths)
+            {
+                var methods = new List<string>();
+                for (var i = 0; i < MAX_ALLOWED_HTTP_METHODS; i++)
+                {
+                    var methodKey = key.Replace("Match:Path", $"Match:Methods:{i}");
+                    if (!allConfigs.TryGetValue(methodKey, out var config))
+                        continue;
+
+                    if (config != null)
+                        methods.Add(config);
+                }
+
+                if (!validRoutes.ContainsKey(routeValue))
+                {
+                    validRoutes.Add(routeValue, methods);
+                }
+                else
+                {
+                    validRoutes[routeValue].AddRange(methods);
+                }
+            }
+
+            return validRoutes;
+        }
+
+        private static IReadOnlyList<OperationType> CheckSwaggerDefinitionIsValid(ReverseProxyDocumentFilterConfig.Cluster.Destination.Swagger swagger, IReadOnlyDictionary<string, List<string>> publishedPaths, KeyValuePair<string, OpenApiPathItem> path)
+        {
+            var pathKey = $"{swagger.PrefixPath}{path.Key}";
+            if (!publishedPaths.TryGetValue(pathKey, out var methods))
+            {
+                return new List<OperationType>();
+            }
+
+            var operations = methods.Select(q =>
+            {
+                switch (q)
```

Comment:
Can you make cases as ReadOnlyDictionary field please? So you can delete const and use Count and get OperationType by key.

Reply:
changed.